eclipse-cdt-cloud/vscode-memory-inspector
# Implement Options Widget & Track Active Session

#### What it does


Fixes #6, Closes #7, Resolves #8.

- Modifies the Memory Provider to maintain a list of capabilities for each active debug session so that it can behave appropriately according to the capabilities of that session.
- Splits the `validDebugger` context key into `canRead` and `canWrite`, since reading and writing capabilities are separate.
- Changes the font for the data and address display to the `--vscode-editor-font-family` so that it is a monospace font.
- Adds a view to the panel that controls the memory parameters (reference, offset, count) and the render parameters (bytes/group, groups/row, endianness) and updates the view appropriately.

#### How to test


1. Access some memory and confirm that the data (but not column headers) are displayed in a monospace font.
---
1. Run two debug sessions simultaneously, one that can read memory and one that cannot - try starting them in different orders.
2. Confirm that the context menu items are available for variables in the session that can read memory but not for variables in the session that cannot.
---
1. Open a memory view
2. Use the configuration view to change the parameters of the memory fetch.
3. Confirm that clicking `Go` fetches memory that matches the parameters entered.
4. With memory loaded, modify the render parameters and confirm that the view behaves as expected.

#### Review checklist

To be continued...


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
